pkg/lib/license/license_validator.go

```diff
@@ -0,0 +1,113 @@
+package license
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/MicahParks/keyfunc/v3"
+	"github.com/golang-jwt/jwt/v5"
+)
+
+// LicenseValidator handles JWT validation with JWKS support
+type LicenseValidator struct {
+	keyFunc jwt.Keyfunc
+}
+
+// LicenseClaims represents the expected claims in your license token
+type LicenseClaims struct {
+	jwt.RegisteredClaims
+	// Add your custom license claims here
+	Product        string            `json:"product,omitempty"`
+	LicenseID      string            `json:"license_id,omitempty"`
+	LicenseType    string            `json:"license_type,omitempty"`
+	LicenseVersion string            `json:"license_version,omitempty"`
+	CustomerID     string            `json:"customer_id,omitempty"`
+	Features       []string          `json:"features,omitempty"`
+	Limitations    map[string]string `json:"limitations,omitempty"`
```

Comment:
what is the best way to define number of nodes the license supports?

Reply:
@wdbaruni resolving this one and agreed on adding `capabilities` claim, and removing Features and Limitations.
Capabilities will be a `map[string]string`